### PR TITLE
Threat miner

### PR DESCRIPTION
**Pull request by idobloch**
Integration has the following commands:

1. domain
2. ip
3. file

## Status
Ready

## Related Issues
https://github.com/demisto/etc/issues/14231

## Description
Add integration to ThreatMiner

## Must have
- [x] Tests
- [x] Documentation https://github.com/demisto/etc/issues/14842
- [ ] Code Review

### CODE REVIEW

**Comment by meirwah** [comment]
@anara123 @yuvalbenshalom who can review this PR?

**Comment by yuvalbenshalom** [comment]
@meirwah Ill try to do it today.
BTW, why `Code Review` checkbox is checked then?

**Comment by meirwah** [comment]
@yuvalbenshalom plz don't forget to review this

File: Integrations/integration-ThreatMiner.yml [hunk 1]
```diff
@@ -0,0 +1,863 @@
...
+  if len(threatminerResultsAsArray) == 0:
+    return threatminerResultsAsArray
+
+  return threatminerResultsAsArray[0]
+
+def get_file_whois_and_update_theartMinerFileContext(rawdataFile, context):
+  if len(rawdataFile['raw_whois']) == 0:
+    return rawdataFile['raw_whois']
+
+  context['ThreatMiner']['File']['MD5'] = rawdataFile['raw_whois']['md5']
```
Comment (anara123) [inline Integrations/integration-ThreatMiner.yml]: this should be like:
```
context = 
     'ThreatMiner.File(val.MD5 == obj.MD5)': {
             'MD5': rawdataFile['raw_whois']['md5'],
             'Sha1': rawdataFile['raw_whois']['sha1'],
            etc.....
      }
}
```

File: Integrations/integration-ThreatMiner.yml [hunk 2]
```diff
@@ -0,0 +1,863 @@
...
+
+  type: python
+  commands:
+  - name: domain
+    arguments:
+    - name: domain
+      required: true
+      description: Domain name
+    outputs:
+    - contextPath: ThreatMiner.Domain
```
Comment (anara123) [inline Integrations/integration-ThreatMiner.yml]: You are missing the standard context outputs
```
Domain.Address
Domain.WHOIS.Server
....
```
please take a look here https://github.com/demisto/content/blob/master/docs/context_standards/README.MD

File: Integrations/integration-ThreatMiner.yml [hunk 3]
```diff
@@ -0,0 +1,839 @@
...
+        amountOfDetections = len(fileAntiVirus)
+
+        dbot_scores = get_dbot_score_report(amountOfDetections, hashedFile, fileContext)
+
+        markDownResult = create_file_command_markdown(hashedFile, threatMinerFileContext['ThreatMiner']['File'])
+
+        context = {
+            'ThreatMiner.File(val.MD5 && val.MD5 == obj.MD5)':  createContext(threatMinerFileContext['ThreatMiner']['File'], removeNull=True),
+            'File(val.MD5 && val.MD5 == obj.MD5)': createContext(fileContext, removeNull=True),
+            'DBotScore(val.Vendor && val.Indicator && val.Vendor == obj.Vendor && val.Indicator == obj.Indicator)': createContext(dbot_scores, removeNull=True)
```
[the remark this answers is not in the document]
Comment (idobloch) [reply Integrations/integration-ThreatMiner.yml]: I am not sure what is DT? Do you mean I shouldn't use createContext() in DBotScore?

File: Integrations/integration-ThreatMiner.yml (hunk 1 above)
Comment (idobloch) [reply Integrations/integration-ThreatMiner.yml]: I've tied the MD5 values but just outside the context function. Do I have to do it inside?

File: Integrations/integration-ThreatMiner.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (anara123) [reply Integrations/integration-ThreatMiner.yml]: DT filter is what you use inside of parenthesis  `(val.Vendor && val.Indicator && val.Vendor == obj.Vendor && val.Indicator == obj.Indicator)`.
Just remove the parenthesis, leave just the DBotScore

**Comment by idobloch** [comment]
I've finished all the requested changes, so you can continue the PR review.
Thanks

**Comment by meirwah** [comment]
@idobloch can you resolve conflicts here?

**Comment by meirwah** [comment]
@anara123 @yuvalbenshalom plz approve if all changes asked are met

**Comment by anara123** [comment]
@idobloch 
```
Test playbook: ThreatMiner-Test with integration(s): ThreatMiner start ------
integration was not found
Failed to create instance
Failed: playbook: ThreatMiner-Test with integration(s): ThreatMiner failed
```

**Comment by meirwah** [comment]
@idobloch are we done here? @anara123 can we merge?

**Pull request merged**